### PR TITLE
Promote kubelet config errors

### PR DESCRIPTION
**Pull request by cynepco3hahue**
Promote kubelet config errors to the performance profile. It will promote only the failure error only when it the most recent condition.

### CODE REVIEW

File: controllers/performanceprofile_controller_test.go [hunk 1]
```diff
@@ -175,7 +175,100 @@ var _ = Describe("Controller", func() {
...
+			}
+			Expect(r.Get(context.TODO(), key, updatedProfile)).ToNot(HaveOccurred())
+
+			degradedCondition := conditionsv1.FindStatusCondition(updatedProfile.Status.Conditions, conditionsv1.ConditionDegraded)
+			Expect(degradedCondition.Status).To(Equal(corev1.ConditionTrue))
+			Expect(degradedCondition.Message).To(Equal("Test failure condition"))
+			Expect(degradedCondition.Reason).To(Equal(conditionKubeletFailed))
+		})
+
+		It("should promote not promote old failure condition", func() {
```
Comment (fromanirh) [inline controllers/performanceprofile_controller_test.go:228]: nit: "promote not promote" -> "not promote"

**Review by fromanirh — APPROVED** [review]
/lgtm
/hold
just a small nit inline

File: controllers/performanceprofile_controller_test.go (hunk 1 above)
Comment (cynepco3hahue) [reply controllers/performanceprofile_controller_test.go]: done

**Review by fromanirh — APPROVED** [review]
/lgtm
/approve

**Comment by fromanirh** [comment]
```
Unable to connect to the server: dial tcp: lookup api.ci-op-39vrrt08-24cc7.origin-ci-int-gce.dev.openshift.com on 172.30.0.10:53: no such host
```
:\

**Comment by fromanirh** [comment]
failed for timeout. Let's retry

**Comment by fromanirh** [comment]
/hold cancel
nit resolved

**Pull request merged**